specs.js

```diff
@@ -0,0 +1,149 @@
+      expect(trim.getDocument).to.be.a('function')
+    })
+
+    it('should getDocument succesfully', function (done) {
+      var documentId = 'someIDthatPointsToADoucment'
+      var getDocumentMock = nock(mockURL)
+        .get('/get')
+        .query({id: documentId, securityToken: mockToken})
+        .reply(200, {
+          msg: 'response' // TODO what shape object does getDocument actually return
+        })
+
+      trim.getDocument(documentId, function (err, data) {
+        expect(err).to.be.an('null')
+        expect(data).not.to.be.an('undefined')
+
+        getDocumentMock.done()
+        done()
+      })
+    })
+  })
+
+  describe('when using the createContainer function', function () {
+    it('should exist on the trim instance', function () {
+      expect(trim.createContainer).to.be.a('function')
+    })
+
+    it('should be able to create a container', function (done) {
+      var createContainerMock = nock(mockURL)
+        .filteringRequestBody(function (body) {
+          return '*' // this allows us to ignore the contents of the post for this asertion
+        })
+        .post('/CreateContainer', '*')
+        .query({ securityToken: mockToken })
+        .reply(201, {
+          msg: 'create response' // TODO determine the shape of the response
+        })
+
+      trim.createContainer('someFolderName', 'somePrivacySetting', 'someParentFolder', function (err, data) {
+        expect(err).to.be.a('null')
+        expect(data).not.to.be.a('undefined')
+
+        createContainerMock.done()
+        done()
+      })
+
+    })
+  })
+
+  describe('when using the createRecord function', function () {
+    it('should exist on the trim instance', function () {
+      expect(trim.createRecord).to.be.a('function')
+    })
+
+    it('should be able to succesfully use the function', function (done) {
+      var createRecordMock = nock(mockURL)
+        .filteringRequestBody(function (bodyt) {
+          return '*'
+        })
+        .post('/AddRecordToTrim', '*')
+        .query({ securityToken: mockToken })
+        .reply(201, { RecordNo: '123456' })
+
+      trim.createRecord('newTitle', 'containerID', 'extensiontype', 'fileData', function (err, data) {
+        expect(err).to.be.an('null')
+        expect(data).not.to.be.an('undefined')
+
+        expect(data).to.be.a('string')
+        expect(data).to.equal('123456')
+        createRecordMock.done()
+        done()
+      })
+    })
+
+    it('should fail when TRIM responds without a record number number', function (done) {
+      var createRecordMock = nock(mockURL)
+        .filteringRequestBody(function (bodyt) {
+          return '*'
+        })
+        .post('/AddRecordToTrim', '*')
+        .query({ securityToken: mockToken })
+        .reply(500, { msg: 'it failed wtfomg' }) // determine shape of response
+
+      trim.createRecord('newTitle', 'containerID', 'extensiontype', 'fileData', function (err, data) {
+        expect(err).not.to.be.an('null')
```

Comment:
This could be `expect(err).not.to.be.null` or `expect(err).to.exist` which might be nicer. `.exist` checks for `!null` or `!undefined` apparently.